src/iguana/algorithms/clas12/RGAFiducialFilter/Algorithm.cc

```diff
@@ -0,0 +1,551 @@
+#include "Algorithm.h" 
+
+#include <yaml-cpp/yaml.h>
+
+#include <algorithm>
+#include <cmath>
+#include <cstdlib>
+#include <limits>
+#include <map>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <fstream>
+#include <vector>
+
+// Default settings defined and adjustable in Config.yaml (CAL)
+//
+// * Forward Tagger cut on annulus and missing holes for pid==11,22
+//
+// * Calorimeter cut on lv,lw>9,13.5,18 depending on SetStrictness([default = 1]) for pid==11,22; 
+//    Strictness 1 loosely recommended for spin-asymmetries, cross sections may want to consider
+//    higher strictness as well as ...
+//    !!! TO DO: implement run-by-run data/MC matching for missing PMTs, etc.
+//
+// * Central Detector cut on areas between 3 sectors and require tracks inside detector (edge > 0);
+//    for all charged hadrons
+// 
+// * Drift chamber cut on region 1 > 3 cm, region 2 > 3, region3 > 10; inbending tracks also require
+//    region 1 > 10, region 2 > 10 for theta < 10; cuts for all charged leptons and hadrons
+
+namespace iguana::clas12 {
+static bool banklist_has(hipo::banklist& banks, const char* name) {
```

Comment:
The CI tests aren't running for this; adding `REGISTER_IGUANA_ALGORITHM` here should fix that:

```suggestion
namespace iguana::clas12 {

REGISTER_IGUANA_ALGORITHM(RGAFiducialFilter, "clas12::RGAFiducialFilter");

static bool banklist_has(hipo::banklist& banks, const char* name) {
```

I'll take a closer look at this PR later this week; for now, let's see if the CI tests pass (they run on every new commit you push).